src/validation.cpp

```diff
@@ -4299,7 +4299,7 @@ bool ChainstateManager::AcceptBlock(const std::shared_ptr<const CBlock>& pblock,
         }
         ReceivedBlockTransactions(block, pindex, blockPos);
     } catch (const std::runtime_error& e) {
-        return FatalError(GetNotifications(), state, std::string("System error: ") + e.what());
+        return FatalError(GetNotifications(), state, strprintf(_("System error: %s"), e.what()));
```

Comment:
nit: while touching, would be useful to consistently add context (+[here](https://github.com/bitcoin/bitcoin/pull/29672/files#diff-97c3a52bc5fad452d82670a7fd291800bae20c7bc35bb82686c2c0a4ea7b5b98R4986))
```suggestion
        return FatalError(GetNotifications(), state, strprintf(_("System error while writing block to history file: %s"), e.what()));
```